src/raft/raw_node.rs

```diff
@@ -124,6 +131,9 @@ impl Ready {
         if raft.raft_log.get_unstable().snapshot.is_some() {
             rd.snapshot = raft.raft_log.get_unstable().snapshot.clone().unwrap();
         }
+        if !raft.read_states.is_empty() {
+            rd.read_states = raft.read_states.drain(..).collect();
```

Comment:
What is read only struct? `raft.read_states` will be cleared when `drain` finish. `read_states` is ready to be read out, but it doesn't have to be committed.